src/utils/pathUtil.ts

```diff
@@ -150,12 +152,48 @@ export class PathUtil {
 	 * @param {string} file path to test.
 	 * @returns {vscode.WorkspaceFolder | undefined} the workspace it belongs to
 	 */
-	public static PathExistsRelativeToAnyWorkspace(
+	public static async PathExistsRelativeToAnyWorkspace(
 		file: string
-	): vscode.WorkspaceFolder | undefined {
+	): Promise<vscode.WorkspaceFolder | undefined> {
 		const workspaces = vscode.workspace.workspaceFolders;
-		return workspaces?.find((workspace) =>
-			fs.existsSync(path.join(workspace.uri.fsPath, file))
-		);
+
+		if (!workspaces) {
+			return undefined;
+		}
+
+		const promises = workspaces.map((workspace) =>
+			PathUtil.FileExistsStat(path.join(workspace.uri.fsPath, file)));
+		const idx = (await Promise.all(promises)).findIndex((elem) => elem.exists);
+		if (idx === -1) {
+			return undefined;
+		}
+		return workspaces[idx];
+	}
+
+	/**
+	 * Promisify-ed version of fs.stat.
+	 * @param file
+	 * @returns object containing exists and stat info
+	 */
+	public static async FileExistsStat(file: string): Promise<{ exists: boolean, stat: fs.Stats }> {
+		return new Promise((resolve) => {
+			fs.stat(file, (err, stat) => {
+				resolve({ exists: (err === null), stat });
+			});
+		});
+	}
+
+	/**
+	 * Promisify-ed version of fs.readFile.
+	 * Reads in utf-8 encoding.
+	 * @param file
+	 * @returns file contents
+	 */
+	public static async FileRead(file: string): Promise<string> {
+		return new Promise((resolve) => {
+			fs.readFile(file, 'utf-8', (err, data) => {
+				resolve(err ? '' : data.toString());
+			});
+		});

```

Comment:
Btw, `fs` exposes a Promises interface (since Node 14 I think?).

I usually `import { promises as fs } from 'fs'`, then you get promise versions of most fs functions.